graphix/gflow.py

```diff
@@ -245,123 +140,71 @@ def find_flow(
 
     Parameters
     ----------
-    graph: :class:`networkx.Graph`
-        Graph (incl. input and output)
-    iset: set
-        set of node labels for input
-    oset: set
-        set of node labels for output
-    meas_planes: dict(int, Plane)
-        measurement planes for each qubits. meas_planes[i] is the measurement plane for qubit i.
-        Note that an underlying graph has a causal flow only if all measurement planes are Plane.XY.
-        If not specified, all measurement planes are interpreted as Plane.XY.
+    graph : `networkx.Graph`
+        The underlying graph.
+    iset : `collections.abc.Set`
+        Input nodes.
+    oset : `collections.abc.Set`
+        Output nodes.
 
     Returns
     -------
-    f: list of nodes
-        causal flow function. f[i] is the qubit to be measured after qubit i.
-    l_k: dict
-        layers obtained by gflow algorithm. l_k[d] is a node set of depth d.
+    Flow function and layer if found, otherwise `None`.
     """
-    check_meas_planes(meas_planes)
-    nodes = set(graph.nodes)
-    edges = set(graph.edges)
-
-    if meas_planes is None:
-        meas_planes = dict.fromkeys(nodes - oset, Plane.XY)
+    # meas is left undocumented because it is essentially redundant.
+    if meas is None:
+        meas = _default_construct(graph.nodes - oset)
+    if any(_p_convert(v) != Plane_.XY for v in meas.values()):
+        return None
+    return flow_module.find(graph, iset, oset)
 
-    for plane in meas_planes.values():
-        if plane != Plane.XY:
-            return None, None
 
-    l_k = dict.fromkeys(nodes, 0)
-    f = {}
-    k = 1
-    v_c = oset - iset
-    return flowaux(nodes, edges, iset, oset, v_c, f, l_k, k)
+def find_gflow(
+    graph: nx.Graph[int],
+    iset: AbstractSet[int],
+    oset: AbstractSet[int],
+    meas: Mapping[int, Measurement] | None = None,
```

Comment:
As for `find_flow`, I would prefer to retain the earlier version `Mapping[int, Plane]`, and perform the conversion in `generate_from_graph`.